posts/2021-10-08-nodejs-read-file-line-by-line.md

```diff
@@ -4,33 +4,34 @@ title: 4 ways to read file line by line in Node.js
 date: 2021-10-08T21:30:35.000+11:00
 comments: true
 tags:
-- Software Engineering
-- Javascript
-- NodeJs
+  - Software Engineering
+  - Javascript
+  - NodeJs
 cover: "/images/nodejs-read-file-line-by-line/01nodejs-read-file-line-by-line.jpg"
 pagetitle: 4 ways to read file line by line in Node.js
 description: Learn how to read file line by line in Node.js with sync and async methods using native and NPM modules.
 keywords: nodejs read file line by line, read file line by line nodejs, node.js read file line by line, read file line by line node.js
-
 ---
-There are multiple ways to read a file line by line with Node.js. In Node.js files can be read in sync way or in an async way. With the async path, it is possible to read large files without loading all the content of the file into memory. 
+
+There are multiple ways to read a file line by line with Node.js. In Node.js files can be read in sync way or in an async way. With the async path, it is possible to read large files without loading all the content of the file into memory.
 
 <!-- more -->
 
 Reading the whole file at once will make the process memory intensive. With the ability to load and read a file line by line it enables us to stop the process at any step as per need. In this post, we will look into 3 ways to read a file line by line using Node.js with memory usage comparison.
 
 <img class="center" loading="lazy" src="/images/nodejs-read-file-line-by-line/01nodejs-read-file-line-by-line.jpg" title="4 ways to read file line by line with Node.js" alt="4 ways to read file line by line with Node.js">
+
 ## Table of contents
 
-* [Prerequisites](#prerequisites)
-* [The test file](#the-test-file)
-* [Read file sync](#read-file-sync)
-* [Readline](#readline)
-* [N-readlines](#n-readlines)
-* [Line reader](#line-reader)
-* [Other options](#other-options)
-* [Quick comparison](#quick-comparison)
-* [Conclusion](#conclusion)
+- [Prerequisites](#prerequisites)
+- [The test file](#the-test-file)
+- [Read file sync](#read-file-sync)
+- [Readline](#readline)
+- [N-readlines](#n-readlines)
+- [Line reader](#line-reader)
+- [Other options](#other-options)
+- [Quick comparison](#quick-comparison)
+- [Conclusion](#conclusion)
```

Comment:
`*` should be the same IMO.

Reply:
This is because of my formatter('Prettier' vs code extension) . Still works the same as * but I will make sure.